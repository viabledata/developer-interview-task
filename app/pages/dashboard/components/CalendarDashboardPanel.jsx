import React from 'react';
import {withRouter} from "react-router";
import AppConstants from "../../../common/AppConstants";

export class CalendarDashboardPanel extends React.Component {


    calendar(e) {
        e.preventDefault();
        this.props.history.replace({
            pathname: AppConstants.DASHBOARD_PATH,
            shiftPresent: this.props.hasActiveShift
        });
    }
    render() {
        return <li className="__card govuk-grid-column-one-third" id="calendarPanel" style={{marginBottom: '30px'}}>
            <a href={AppConstants.DASHBOARD_PATH} className="card__body" id="calendarPageLink" style={{color: '#005ea5'}} onClick={this.calendar.bind(this)}>
                <span className="govuk-!-font-size-36 govuk-!-font-weight-bold">Calendar</span>
            </a>
            <div className="card__footer">
                <span className="govuk-!-font-size-19">Coming soon</span>
            </div>
        </li>
    }
}

export default withRouter(CalendarDashboardPanel);
