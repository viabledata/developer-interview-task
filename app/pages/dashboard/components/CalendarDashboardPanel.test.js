import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { CalendarDashboardPanel } from './CalendarDashboardPanel';


describe('Calendar Dashboard Panel', () => {
  const mockStore = configureStore();
  let store;
  const initialState = {
    'calendar-page': {},
  };
  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders calendar dashboard panel', async () => {
    const props = {
      hasActiveShift: true,
    };
    const wrapper = await mount(<CalendarDashboardPanel
      store={store}
      {...props}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('navigates to calendar page on click', async () => {
    const history = createMemoryHistory('/calendar');

    const props = {
      history,
      hasActiveShift: true,
    };
    const wrapper = await mount(<Router history={history}>
      <CalendarDashboardPanel
        store={store}
        {...props}
      />
                                </Router>);

    const calendarPageLink = wrapper.find('#calendarPageLink');
    expect(calendarPageLink.exists()).toEqual(true);

    calendarPageLink.simulate('click');
    expect(props.history.location.pathname).toEqual('/calendar');
  });
});
