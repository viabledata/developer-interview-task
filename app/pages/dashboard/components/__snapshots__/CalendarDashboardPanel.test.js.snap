// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Dashboard Panel renders calendar dashboard panel 1`] = `
<CalendarDashboardPanel
  hasActiveShift={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <li
    className="__card govuk-grid-column-one-third"
    id="calendarPanel"
    style={
      Object {
        "marginBottom": "30px",
      }
    }
  >
    <a
      className="card__body"
      href="/calendar"
      id="calendarPageLink"
      onClick={[Function]}
      style={
        Object {
          "color": "#005ea5",
        }
      }
    >
      <span
        className="govuk-!-font-size-36 govuk-!-font-weight-bold"
      >
        Calendar
      </span>
    </a>
    <div
      className="card__footer"
    >
      <span
        className="govuk-!-font-size-19"
      >
        Coming soon
      </span>
    </div>
  </li>
</CalendarDashboardPanel>
`;
